src/main/java/kirby/TaskList.java

```diff
@@ -0,0 +1,136 @@
+package kirby;
+
+import java.util.ArrayList;
+
+import kirby.tasks.Task;
+
+/**
+ * TaskList class handles the list of tasks - adding and deleting tasks from it and other operations.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private int taskCount = 0;
+
+    /**
+     * Constructor of the TaskList class.
+     *
+     * @param tasks List of initial tasks.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        if (tasks == null) {
+            this.tasks = new ArrayList<>();
+        } else {
+            this.tasks = tasks;
+            taskCount = tasks.size();
+        }
+    }
+
+    /**
+     * Returns the number of tasks in the list.
+     *
+     * @return Number of tasks in the list.
+     */
+    public int getTaskCount() {
+        return this.taskCount;
+    }
+
+    /**
+     * Sets a specified task as completed.
+     *
+     * @param taskNumber Index of the task to be marked.
+     */
+    public void setTaskMarked(int taskNumber) {
```

Comment:
Really good naming of setters - I immediately know what the method is used for 👍